[LV] Teach m_One, m_ZeroInt patterns to look through broadcasts

david-arm opened the pull request:
In VPlanPatternMatch.h I have changed the int_pred_ty code to look through broadcasts in order to catch more cases, i.e. multiplying by a splat of one, etc.

lukel97 reviewed (commented):
I thought we already had this simplification in VPlanTransforms::simplifyRecipes, but looking at it now I think the issue is that we have to add the explicit broadcast in between in expandVPWidenIntOrFpInduction because we're after materializeBroadcasts, and that probably breaks the pattern.

I don't suppose its possible to try move materializeBroadcasts later till after convertToConcreteRecipes + the last simplifyRecipes?

lukel97 reviewed (approved):
LGTM

fhahn reviewed (approved):
LGTM, thanks

david-arm committed:
[LV] Teach m_One, m_ZeroInt patterns to look through broadcasts

In VPlanPatternMatch.h I have changed the int_pred_ty code to
look through broadcasts in order to catch more cases, i.e.
multiplying by a splat of one, etc.
